### ISSUE TITLE
19 tests fail

### ISSUE THREAD

**Issue by yurivict** [scikit-build/scikit-build#791]
Please see the log: [scikit-build-test-0.16.1.log](https://github.com/scikit-build/scikit-build/files/9896692/scikit-build-test-0.16.1.log)

**Comment by henryiii** [comment]
Are these new failures compared to 0.15.0?

**Comment by henryiii** [comment]
I do see the "exclude data" tests showing up. That means you have setuptools_scm installed while testing. You need setuptools_scm to build scikit-build, but it should not leak into the runtime requirements. Users should not be forced to install setuptools_scm when they install scikit-build, as setuptools_scm changes the way files are included by setuptools simply by being present.

**Comment by yurivict** [comment]
When ```setuptools_scm``` has been removed now 13 tests fail.

> Are these new failures compared to 0.15.0?

0.15.0 also had 13 failed tests.

**Comment by henryiii** [comment]
Excellent, as long as we aren't regressing. I'd like to work on the failures, but first goal is to make sure they don't grow. :) Do you know why there's no failed test summary at the end (`-ra` is specified in the pyproject.toml file)?

**Comment by henryiii** [comment]
Here's my grep-based summary (of the 17 tests):

```
tests/test_broken_project.py::test_invalid_generator FAILED              [  2%]
tests/test_hello_cpp.py::test_hello_develop FAILED                       [ 22%]
tests/test_include_exclude_data.py::test_hello_sdist FAILED              [ 26%]
tests/test_include_exclude_data.py::test_hello_sdist_with_base FAILED    [ 27%]
tests/test_issue274_support_default_package_dir.py::test_install_command FAILED [ 27%]
tests/test_issue274_support_default_package_dir.py::test_test_command FAILED [ 28%]
tests/test_issue274_support_one_package_without_package_dir.py::test_install_command FAILED [ 28%]
tests/test_issue274_support_one_package_without_package_dir.py::test_test_command FAILED [ 28%]
tests/test_issue334_configure_cmakelists_non_cp1252_encoding.py::test_install_command FAILED [ 29%]
tests/test_issue401_sdist_with_symlinks.py::test_sdist_with_symlinks FAILED [ 34%]
tests/test_manifest_in.py::test_manifest_in_sdist FAILED                 [ 35%]
tests/test_numpy.py::test_pep518_findpython FAILED                       [ 36%]
tests/test_pep518.py::test_pep518 FAILED                                 [ 38%]
tests/test_pep518.py::test_dual_pep518 FAILED                            [ 38%]
tests/test_platform.py::test_fortran_compiler FAILED                     [ 40%]
tests/test_platform.py::test_generator_cleanup FAILED                    [ 41%]
tests/test_setup.py::test_setup_requires_keyword_include_cmake FAILED    [ 50%]
tests/test_skbuild.py::test_invalid_generator[generator_args0] FAILED    [ 87%]
tests/test_skbuild.py::test_invalid_generator[generator_args1] FAILED    [ 87%]
```

Looks like Fortran is missing, I don't know why the "invalid generator" tests are broken, but they are somewhat low priority. You are getting permission errors, like `'/usr/local/lib/python3.9/site-packages/test-easy-install-90201.write-test'` for example. (I don't care about easy-install working, it's deprecated and we will be removing it soon - maybe all the erroring ones are to to deprecated commands?). And the "PEP 518" failures are from isolated builds not working; that's the one I'd worry about the most.

```
Looking in links: /tmp/pytest-of-yuri/pytest-530/wheelhouse0
Processing /tmp/pytest-of-yuri/pytest-530/wheelhouse0/scikit_build-0.16.1-py3-none-any.whl
Processing /tmp/pytest-of-yuri/pytest-530/wheelhouse0/setuptools-65.5.0-py3-none-any.whl
Processing /tmp/pytest-of-yuri/pytest-530/wheelhouse0/numpy-1.23.4.tar.gz
  Installing build dependencies: started
  Installing build dependencies: finished with status 'error'
  error: subprocess-exited-with-error
  
  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> [4 lines of output]
      Looking in links: /tmp/pytest-of-yuri/pytest-530/wheelhouse0, /tmp/pytest-of-yuri/pytest-530/wheelhouse0
      Ignoring packaging: markers 'platform_machine == "arm64"' don't match your environment
      ERROR: Could not find a version that satisfies the requirement setuptools==59.2.0 (from versions: 65.5.0)
      ERROR: No matching distribution found for setuptools==59.2.0
      [end of output]
```

**Comment by yurivict** [comment]
> Do you know why there's no failed test summary at the end (-ra is specified in the pyproject.toml file)?

FreeBSD ports framework runs tests in its own way: https://cgit.freebsd.org/ports/tree/Mk/Uses/python.mk#n810

**Comment by henryiii** [comment]
Ahh, okay. I'd recommend `-ra` over `-rs`, a summary of failures is really useful, just as or more useful than a summary of skips (and you get both with `-ra`). But that explains it. I'll try to address some of the failing tests in the next release. They mostly look like things that are not likely to break users (with the partial possible exception of the PEP 518 one).

**Comment by henryiii** [comment]
NumPy pins setuptools exactly, which breaks this test. They should not be doing this, but they do. You don't have a binary version of NumPy on your platform, so it's triggering the build, the build can't find numpy's exactly pinned setuptools, so it breaks.

**Comment by henryiii** [comment]
I think the best thing to do is to avoid numpy for PEP 518 tests. I'll see if we can drop it.

**Comment by hroncok** [comment]
We are hitting several similar issues when trying to package this into Fedora.

The following tests invoke easy_install from within and try to write to `/usr/lib64/python3.11/site-packages` -- hence we cannot run them in our environment, because we get PermissionErrors. I suspect the tests only work if Python is executed from within a virtual environment (or maybe if the tests run as root, which is a bad idea anyway). Ideally, the tests would use pytest fixture that creates the virtual environment in tmpdir.

```
FAILED tests/test_hello_cpp.py::test_hello_develop - SystemExit: error: can't...
FAILED tests/test_issue274_support_default_package_dir.py::test_install_command
FAILED tests/test_issue274_support_default_package_dir.py::test_test_command
FAILED tests/test_issue274_support_one_package_without_package_dir.py::test_install_command
FAILED tests/test_issue274_support_one_package_without_package_dir.py::test_test_command
FAILED tests/test_issue334_configure_cmakelists_non_cp1252_encoding.py::test_install_command
```

The following tests probably [only work when setuptools_scm is **not** installed](https://github.com/scikit-build/scikit-build/issues/791#issuecomment-1296487334) which is impossible in our environment, as we use it to build the wheel and we cannot uninstall packages during the build. Ideally, the tests would amend their expectations to pass even when setuptools_scm changes the result somehow.

```
FAILED tests/test_include_exclude_data.py::test_hello_sdist - AssertionError:...
FAILED tests/test_include_exclude_data.py::test_hello_sdist_with_base - Asser...
FAILED tests/test_issue401_sdist_with_symlinks.py::test_sdist_with_symlinks
FAILED tests/test_manifest_in.py::test_manifest_in_sdist - AssertionError: as...
```

The following tests require internet access to pull in PEP 518 build dependencies and our build is offline:

```
ERROR tests/test_numpy.py::test_pep518_findpython - subprocess.CalledProcessE...
ERROR tests/test_pep518.py::test_pep518 - subprocess.CalledProcessError: Comm...
ERROR tests/test_pep518.py::test_dual_pep518 - subprocess.CalledProcessError:...
```

The rest of the tests passed for us when we pulled in git, cmake, ninja-build, gcc, gcc-c++ and gcc-fortran.

cc @hrnciar

**Comment by henryiii** [comment]
Could you turn off the deprecated tests? `-m "not deprecated"` will turn off all the "command" tests; they test distutils/setuptools commands, some of which write directly to site-packages. We won't be supporting those long term anyway, so haven't worried much about fixing the tests by adding venvs (which is tricky to do properly). I need to add a similar marker for PEP 518 as well, as they don't work in all environments (like conda-forge) - I think `-k "not pep518"` works for now.

The setuptools_scm one is tricky - adding setuptools_scm to your environment changes all setuptools builds by completely changing file discovery, regardless of whether the package expected it or not. It's important that the final package doesn't include setuptools_scm as an install dependency. As long as that's true, it's safe to skip these tests - maybe we could add a marker?

**Comment by henryiii** [comment]
Xref: #834

**Comment by henryiii** [comment]
After #837, then this will be:

```
-m "not deprecated and not isolated and not nosetuptoolsscm"
```

Conda-forge should be able to use this too (minus the nosetuptoolsscm one).

**Comment by yurivict** [comment]
0.16.6 has these filures:

```
================================================================================================ ERRORS =================================================================================================
_______________________________________________________________________ ERROR at setup of test_invalid_cmake[CalledProcessError] ________________________________________________________________________
file /usr/ports/devel/py-scikit-build/work-py39/scikit-build-0.16.6/tests/test_broken_project.py, line 80
  @pytest.mark.parametrize("exception", [CalledProcessError, OSError])
  def test_invalid_cmake(exception, mocker):
E       fixture 'mocker' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, pep518, pep518_wheelhouse, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, testrun_uid, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.

/usr/ports/devel/py-scikit-build/work-py39/scikit-build-0.16.6/tests/test_broken_project.py:80
_____________________________________________________________________________ ERROR at setup of test_invalid_cmake[OSError] _____________________________________________________________________________
file /usr/ports/devel/py-scikit-build/work-py39/scikit-build-0.16.6/tests/test_broken_project.py, line 80
  @pytest.mark.parametrize("exception", [CalledProcessError, OSError])
  def test_invalid_cmake(exception, mocker):
E       fixture 'mocker' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, pep518, pep518_wheelhouse, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, testrun_uid, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.

/usr/ports/devel/py-scikit-build/work-py39/scikit-build-0.16.6/tests/test_broken_project.py:80
____________________________________________________________________________ ERROR at setup of test_first_invalid_generator _____________________________________________________________________________
file /usr/ports/devel/py-scikit-build/work-py39/scikit-build-0.16.6/tests/test_broken_project.py, line 112
  def test_first_invalid_generator(mocker, capfd):
E       fixture 'mocker' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, pep518, pep518_wheelhouse, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, testrun_uid, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.

/usr/ports/devel/py-scikit-build/work-py39/scikit-build-0.16.6/tests/test_broken_project.py:112
_______________________________________________________________________________ ERROR at setup of test_invalid_generator ________________________________________________________________________________
file /usr/ports/devel/py-scikit-build/work-py39/scikit-build-0.16.6/tests/test_broken_project.py, line 134
  def test_invalid_generator(mocker, capfd):
E       fixture 'mocker' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, monkeypatch, no_cover, pep518, pep518_wheelhouse, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, testrun_uid, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.

/usr/ports/devel/py-scikit-build/work-py39/scikit-build-0.16.6/tests/test_broken_project.py:134
=============================================================================================== FAILURES ================================================================================================
__________________________________________________________________________________________ test_hello_develop ___________________________________________________________________________________________

self = <setuptools.command.develop.develop object at 0x930aee550>

    def check_site_dir(self):  # noqa: C901  # is too complex (12)  # FIXME
        """Verify that self.install_dir is .pth-capable dir, if needed"""
    
        instdir = normalize_path(self.install_dir)
        pth_file = os.path.join(instdir, 'easy-install.pth')
    
        if not os.path.exists(instdir):
            try:
                os.makedirs(instdir)
            except (OSError, IOError):
                self.cant_write_to_target()
    
        # Is it a configured, PYTHONPATH, implicit, or explicit site dir?
        is_site_dir = instdir in self.all_site_dirs
    
        if not is_site_dir and not self.multi_version:
            # No?  Then directly test whether it does .pth file processing
            is_site_dir = self.check_pth_processing()
        else:
            # make sure we can write to target dir
            testfile = self.pseudo_tempname() + '.write-test'
            test_exists = os.path.exists(testfile)
            try:
                if test_exists:
                    os.unlink(testfile)
>               open(testfile, 'w').close()
E               PermissionError: [Errno 13] Permission denied: '/usr/local/lib/python3.9/site-packages/test-easy-install-14104.write-test'

/usr/local/lib/python3.9/site-packages/setuptools/command/easy_install.py:477: PermissionError

During handling of the above exception, another exception occurred:

dist = <skbuild.setuptools_wrap.setup.<locals>.BinaryDistribution object at 0x932d97790>

    def run_commands(dist):
        """Given a Distribution object run all the commands,
        raising ``SystemExit`` errors in the case of failure.
    
        This function assumes that either ``sys.argv`` or ``dist.script_args``
        is already set accordingly.
        """
        try:
>           dist.run_commands()

/usr/local/lib/python3.9/site-packages/setuptools/_distutils/core.py:193: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <skbuild.setuptools_wrap.setup.<locals>.BinaryDistribution object at 0x932d97790>

    def run_commands(self):
        """Run each command that was seen on the setup script command line.
        Uses the list of commands found and cache of command objects
        created by 'get_command_obj()'.
        """
        for cmd in self.commands:
>           self.run_command(cmd)

/usr/local/lib/python3.9/site-packages/setuptools/_distutils/dist.py:968: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <skbuild.setuptools_wrap.setup.<locals>.BinaryDistribution object at 0x932d97790>, command = 'develop'

    def run_command(self, command):
        self.set_defaults()
        # Postpone defaults until all explicit configuration is considered
        # (setup() args, config files, command line and plugins)
    
>       super().run_command(command)

/usr/local/lib/python3.9/site-packages/setuptools/dist.py:1217: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <skbuild.setuptools_wrap.setup.<locals>.BinaryDistribution object at 0x932d97790>, command = 'develop'

    def run_command(self, command):
        """Do whatever it takes to run a command (including nothing at all,
        if the command has already been run).  Specifically: if we have
        already created and run the command named by 'command', return
        silently without doing anything.  If the command named by 'command'
        doesn't even have a command object yet, create one.  Then invoke
        'run()' on that command object (or an existing one).
        """
        # Already been here, done that? then return silently.
        if self.have_run.get(command):
            return
    
        log.info("running %s", command)
        cmd_obj = self.get_command_obj(command)
>       cmd_obj.ensure_finalized()

/usr/local/lib/python3.9/site-packages/setuptools/_distutils/dist.py:986: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <setuptools.command.develop.develop object at 0x930aee550>

    def ensure_finalized(self):
        if not self.finalized:
>           self.finalize_options()

/usr/local/lib/python3.9/site-packages/setuptools/_distutils/cmd.py:107: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <setuptools.command.develop.develop object at 0x930aee550>

    def finalize_options(self):
        ei = self.get_finalized_command("egg_info")
        if ei.broken_egg_info:
            template = "Please rename %r to %r before using 'develop'"
            args = ei.egg_info, ei.broken_egg_info
            raise DistutilsError(template % args)
        self.args = [ei.egg_name]
    
>       easy_install.finalize_options(self)

/usr/local/lib/python3.9/site-packages/setuptools/command/develop.py:52: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <setuptools.command.develop.develop object at 0x930aee550>

    def finalize_options(self):  # noqa: C901  # is too complex (25)  # FIXME
        self.version and self._render_version()
    
        py_version = sys.version.split()[0]
    
        self.config_vars = dict(sysconfig.get_config_vars())
    
        self.config_vars.update({
            'dist_name': self.distribution.get_name(),
            'dist_version': self.distribution.get_version(),
            'dist_fullname': self.distribution.get_fullname(),
            'py_version': py_version,
            'py_version_short': f'{sys.version_info.major}.{sys.version_info.minor}',
            'py_version_nodot': f'{sys.version_info.major}{sys.version_info.minor}',
            'sys_prefix': self.config_vars['prefix'],
            'sys_exec_prefix': self.config_vars['exec_prefix'],
            # Only python 3.2+ has abiflags
            'abiflags': getattr(sys, 'abiflags', ''),
            'platlibdir': getattr(sys, 'platlibdir', 'lib'),
        })
        with contextlib.suppress(AttributeError):
            # only for distutils outside stdlib
            self.config_vars.update({
                'implementation_lower': install._get_implementation().lower(),
                'implementation': install._get_implementation(),
            })
    
        # pypa/distutils#113 Python 3.9 compat
        self.config_vars.setdefault(
            'py_version_nodot_plat',
            getattr(sys, 'windir', '').replace('.', ''),
        )
    
        self.config_vars['userbase'] = self.install_userbase
        self.config_vars['usersite'] = self.install_usersite
        if self.user and not site.ENABLE_USER_SITE:
            log.warn("WARNING: The user site-packages directory is disabled.")
    
        self._fix_install_dir_for_user_site()
    
        self.expand_basedirs()
        self.expand_dirs()
    
        self._expand(
            'install_dir', 'script_dir', 'build_directory',
            'site_dirs',
        )
        # If a non-default installation directory was specified, default the
        # script directory to match it.
        if self.script_dir is None:
            self.script_dir = self.install_dir
    
        if self.no_find_links is None:
            self.no_find_links = False
    
        # Let install_dir get set by install_lib command, which in turn
        # gets its info from the install command, and takes into account
        # --prefix and --home and all that other crud.
        self.set_undefined_options(
            'install_lib', ('install_dir', 'install_dir')
        )
        # Likewise, set default script_dir from 'install_scripts.install_dir'
        self.set_undefined_options(
            'install_scripts', ('install_dir', 'script_dir')
        )
    
        if self.user and self.install_purelib:
            self.install_dir = self.install_purelib
            self.script_dir = self.install_scripts
        # default --record from the install command
        self.set_undefined_options('install', ('record', 'record'))
        self.all_site_dirs = get_site_dirs()
        self.all_site_dirs.extend(self._process_site_dirs(self.site_dirs))
    
        if not self.editable:
>           self.check_site_dir()

/usr/local/lib/python3.9/site-packages/setuptools/command/easy_install.py:294: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <setuptools.command.develop.develop object at 0x930aee550>

    def check_site_dir(self):  # noqa: C901  # is too complex (12)  # FIXME
        """Verify that self.install_dir is .pth-capable dir, if needed"""
    
        instdir = normalize_path(self.install_dir)
        pth_file = os.path.join(instdir, 'easy-install.pth')
    
        if not os.path.exists(instdir):
            try:
                os.makedirs(instdir)
            except (OSError, IOError):
                self.cant_write_to_target()
    
        # Is it a configured, PYTHONPATH, implicit, or explicit site dir?
        is_site_dir = instdir in self.all_site_dirs
    
        if not is_site_dir and not self.multi_version:
            # No?  Then directly test whether it does .pth file processing
            is_site_dir = self.check_pth_processing()
        else:
            # make sure we can write to target dir
            testfile = self.pseudo_tempname() + '.write-test'
            test_exists = os.path.exists(testfile)
            try:
                if test_exists:
                    os.unlink(testfile)
                open(testfile, 'w').close()
                os.unlink(testfile)
            except (OSError, IOError):
>               self.cant_write_to_target()

/usr/local/lib/python3.9/site-packages/setuptools/command/easy_install.py:480: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <setuptools.command.develop.develop object at 0x930aee550>

    def cant_write_to_target(self):
        msg = self.__cant_write_msg % (sys.exc_info()[1], self.install_dir,)
    
        if not os.path.exists(self.install_dir):
            msg += '\n' + self.__not_exists_id
        else:
            msg += '\n' + self.__access_msg
>       raise DistutilsError(msg)
E       distutils.errors.DistutilsError: can't create or remove files in install directory
E       
E       The following error occurred while trying to add or remove files in the
E       installation directory:
E       
E           [Errno 13] Permission denied: '/usr/local/lib/python3.9/site-packages/test-easy-install-14104.write-test'
E       
E       The installation directory you specified (via --install-dir, --prefix, or
E       the distutils default setting) was:
E       
E           /usr/local/lib/python3.9/site-packages/
E       
E       Perhaps your account does not have write access to this directory?  If the
E       installation directory is a system-owned directory, you may need to sign in
E       as the administrator or "root" account.  If you do not have administrative
E       access to this machine, you may wish to choose a different installation
E       directory, preferably one that is listed in your PYTHONPATH environment
E       variable.
E       
E       For information on other options, you may wish to consult the
E       documentation at:
E       
E         https://setuptools.pypa.io/en/latest/deprecated/easy_install.html
E       
E       Please make the appropriate changes for your system and try again.

/usr/local/lib/python3.9/site-packages/setuptools/command/easy_install.py:540: DistutilsError

During handling of the above exception, another exception occurred:

iargs = (), ikwargs = {}

    @functools.wraps(fun)
    def wrapped(*iargs, **ikwargs):
    
        if wrapped.tmp_dir is None:
            wrapped.tmp_dir = _tmpdir(fun.__name__)
            prepare_project(wrapped.project, wrapped.tmp_dir)
            initialize_git_repo_and_commit(wrapped.tmp_dir, verbose=wrapped.verbose_git)
    
>       with execute_setup_py(wrapped.tmp_dir, wrapped.setup_args, disable_languages_test=disable_languages_test):

tests/__init__.py:259: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.9/contextlib.py:119: in __enter__
    return next(self.gen)
tests/__init__.py:244: in execute_setup_py
    exec(setup_code)
setup.py:3: in <module>
    import setuptools
skbuild/setuptools_wrap.py:754: in setup
    return setuptools.setup(**kw)
/usr/local/lib/python3.9/site-packages/setuptools/__init__.py:87: in setup
    return distutils.core.setup(**attrs)
/usr/local/lib/python3.9/site-packages/setuptools/_distutils/core.py:177: in setup
    return run_commands(dist)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

dist = <skbuild.setuptools_wrap.setup.<locals>.BinaryDistribution object at 0x932d97790>

    def run_commands(dist):
        """Given a Distribution object run all the commands,
        raising ``SystemExit`` errors in the case of failure.
    
        This function assumes that either ``sys.argv`` or ``dist.script_args``
        is already set accordingly.
        """
        try:
            dist.run_commands()
        except KeyboardInterrupt:
            raise SystemExit("interrupted")
        except OSError as exc:
            if DEBUG:
                sys.stderr.write("error: %s\n" % (exc,))
                raise
            else:
                raise SystemExit("error: %s" % (exc,))
    
        except (DistutilsError, CCompilerError) as msg:
            if DEBUG:
                raise
            else:
>               raise SystemExit("error: " + str(msg))
E               SystemExit: error: can't create or remove files in install directory
E               
E               The following error occurred while trying to add or remove files in the
E               installation directory:
E               
E                   [Errno 13] Permission denied: '/usr/local/lib/python3.9/site-packages/test-easy-install-14104.write-test'
E               
E               The installation directory you specified (via --install-dir, --prefix, or
E               the distutils default setting) was:
E               
E                   /usr/local/lib/python3.9/site-packages/
E               
E               Perhaps your account does not have write access to this directory?  If the
E               installation directory is a system-owned directory, you may need to sign in
E               as the administrator or "root" account.  If you do not have administrative
E               access to this machine, you may wish to choose a different installation
E               directory, preferably one that is listed in your PYTHONPATH environment
E               variable.
E               
E               For information on other options, you may wish to consult the
E               documentation at:
E               
E                 https://setuptools.pypa.io/en/latest/deprecated/easy_install.html
E               
E               Please make the appropriate changes for your system and try again.

/usr/local/lib/python3.9/site-packages/setuptools/_distutils/core.py:207: SystemExit
----------------------------------------------------------------------------------------- Captured stdout call ------------------------------------------------------------------------------------------
Initialized empty Git repository in /tmp/pytest-of-yuri/pytest-47/test_hello_develop0/.git/
[master (root-commit) f599e18] Initial commit
 11 files changed, 111 insertions(+)
 create mode 100644 CMakeLists.txt
 create mode 100644 bonjour/__init__.py
 create mode 100644 bonjour/data/ciel.txt
 create mode 100644 bonjour/data/soleil.txt
 create mode 100644 bonjour/data/terre.txt
 create mode 100644 bonjourModule.py
 create mode 100644 hello/CMakeLists.txt
 create mode 100644 hello/__init__.py
 create mode 100644 hello/__main__.py
 create mode 100644 hello/_hello.cxx
 create mode 100644 setup.py


--------------------------------------------------------------------------------
-- Trying 'Ninja' generator
--------------------------------
---------------------------
----------------------
-----------------
------------
-------
--
Not searching for unused variables given on the command line.
-- The C compiler identification is Clang 14.0.5
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/libexec/ccache/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- The CXX compiler identification is Clang 14.0.5
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/libexec/ccache/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/pytest-of-yuri/pytest-47/test_hello_develop0/_cmake_test_compile/build
--
-------
------------
-----------------
----------------------
---------------------------
--------------------------------
-- Trying 'Ninja' generator - success
--------------------------------------------------------------------------------

Configuring Project
  Working directory:
    /tmp/pytest-of-yuri/pytest-47/test_hello_develop0/_skbuild/freebsd-13.1-STABLE-amd64-3.9/cmake-build
  Command:
    /usr/local/bin/cmake /tmp/pytest-of-yuri/pytest-47/test_hello_develop0 -G Ninja -DCMAKE_INSTALL_PREFIX:PATH=/tmp/pytest-of-yuri/pytest-47/test_hello_develop0/_skbuild/freebsd-13.1-STABLE-amd64-3.9/cmake-install -DPYTHON_VERSION_STRING:STRING=3.9.16 -DSKBUILD:INTERNAL=TRUE -DCMAKE_MODULE_PATH:PATH=/usr/ports/devel/py-scikit-build/work-py39/scikit-build-0.16.6/skbuild/resources/cmake -DPYTHON_EXECUTABLE:PATH=/usr/local/bin/python3.9 -DPYTHON_INCLUDE_DIR:PATH=/usr/local/include/python3.9 -DPYTHON_LIBRARY:PATH=/usr/local/lib/libpython3.9.so -DPython_EXECUTABLE:PATH=/usr/local/bin/python3.9 -DPython_ROOT_DIR:PATH=/usr/local -DPython_INCLUDE_DIR:PATH=/usr/local/include/python3.9 -DPython_FIND_REGISTRY:STRING=NEVER -DPython_NumPy_INCLUDE_DIRS:PATH=/usr/local/lib/python3.9/site-packages/numpy/core/include -DPython3_EXECUTABLE:PATH=/usr/local/bin/python3.9 -DPython3_ROOT_DIR:PATH=/usr/local -DPython3_INCLUDE_DIR:PATH=/usr/local/include/python3.9 -DPython3_FIND_REGISTRY:STRING=NEVER -DPython3_NumPy_INCLUDE_DIRS:PATH=/usr/local/lib/python3.9/site-packages/numpy/core/include -DCMAKE_BUILD_TYPE:STRING=Release

-- The C compiler identification is Clang 14.0.5
-- The CXX compiler identification is Clang 14.0.5
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/libexec/ccache/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/libexec/ccache/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PythonInterp: /usr/local/bin/python3.9 (found version "3.9.16") 
-- Found PythonLibs: /usr/local/lib/libpython3.9.so (found version "3.9.16") 
-- Performing Test Weak Link MODULE -> SHARED (gnu_ld_ignore) - Success
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/pytest-of-yuri/pytest-47/test_hello_develop0/_skbuild/freebsd-13.1-STABLE-amd64-3.9/cmake-build
[1/3] Building CXX object hello/CMakeFiles/_hello.dir/_hello.cxx.o
[2/3] Linking CXX shared module hello/_hello.cpython-39.so
[2/3] Install the project...
-- Install configuration: "Release"
-- Installing: /tmp/pytest-of-yuri/pytest-47/test_hello_develop0/_skbuild/freebsd-13.1-STABLE-amd64-3.9/cmake-install/./helloModule.py
-- Installing: /tmp/pytest-of-yuri/pytest-47/test_hello_develop0/_skbuild/freebsd-13.1-STABLE-amd64-3.9/cmake-install/hello/_hello.cpython-39.so
-- Installing: /tmp/pytest-of-yuri/pytest-47/test_hello_develop0/_skbuild/freebsd-13.1-STABLE-amd64-3.9/cmake-install/hello/world.py
copying _skbuild/freebsd-13.1-STABLE-amd64-3.9/cmake-install/hello/_hello.cpython-39.so -> hello/_hello.cpython-39.so
copying _skbuild/freebsd-13.1-STABLE-amd64-3.9/cmake-install/hello/world.py -> hello/world.py
copying _skbuild/freebsd-13.1-STABLE-amd64-3.9/cmake-install/helloModule.py -> helloModule.py

running develop
----------------------------------------------------------------------------------------- Captured stderr call ------------------------------------------------------------------------------------------
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint: 
hint: 	git config --global init.defaultBranch <name>
hint: 
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint: 
hint: 	git branch -m <name>
_modinit_prefix:PyInit_
CMake Warning:
  Manually-specified variables were not used by the project:

    Python3_EXECUTABLE
    Python3_FIND_REGISTRY
    Python3_INCLUDE_DIR
    Python3_NumPy_INCLUDE_DIRS
    Python3_ROOT_DIR
    Python_EXECUTABLE
    Python_FIND_REGISTRY
    Python_INCLUDE_DIR
    Python_NumPy_INCLUDE_DIRS
    Python_ROOT_DIR
    SKBUILD


=========================================================================================== warnings summary ============================================================================================
tests/test_command_line.py::test_hide_listing[True-bdist_wheel]
tests/test_command_line.py::test_hide_listing[False-bdist_wheel]
tests/test_filter_manifest.py::test_bdist_wheel_command
tests/test_hello_cpp.py::test_hello_wheel
tests/test_hello_cpp.py::test_hello_wheel
tests/test_hello_cpp.py::test_hello_develop
tests/test_hello_cython.py::test_hello_cython_wheel
tests/test_hello_pure.py::test_hello_pure_wheel
tests/test_include_exclude_data.py::test_include_exclude_data
  /usr/local/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
    warnings.warn(

tests/test_hello_cpp.py::test_hello_develop
  /usr/local/lib/python3.9/site-packages/setuptools/command/easy_install.py:144: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
    warnings.warn(

tests/test_include_exclude_data.py::test_include_exclude_data
  /usr/local/lib/python3.9/site-packages/setuptools/command/build_py.py:153: SetuptoolsDeprecationWarning:     Installing 'hello.data.subdata' as data is deprecated, please list it in `packages`.
      !!
  
  
      ############################
      # Package would be ignored #
      ############################
      Python recognizes 'hello.data.subdata' as an importable package,
      but it is not listed in the `packages` configuration of setuptools.
  
      'hello.data.subdata' has been automatically added to the distribution only
      because it may contain data files, but this behavior is likely to change
      in future versions of setuptools (and therefore is considered deprecated).
  
      Please make sure that 'hello.data.subdata' is included as a package by using
      the `packages` configuration field or the proper discovery methods
      (for example by using `find_namespace_packages(...)`/`find_namespace:`
      instead of `find_packages(...)`/`find:`).
  
      You can read more about "package discovery" and "data files" on setuptools
      documentation page.
  
  
  !!
  
    check.warn(importable)

tests/test_include_exclude_data.py::test_include_exclude_data
  /usr/local/lib/python3.9/site-packages/setuptools/command/build_py.py:153: SetuptoolsDeprecationWarning:     Installing 'hello2.data2.subdata2' as data is deprecated, please list it in `packages`.
      !!
  
  
      ############################
      # Package would be ignored #
      ############################
      Python recognizes 'hello2.data2.subdata2' as an importable package,
      but it is not listed in the `packages` configuration of setuptools.
  
      'hello2.data2.subdata2' has been automatically added to the distribution only
      because it may contain data files, but this behavior is likely to change
      in future versions of setuptools (and therefore is considered deprecated).
  
      Please make sure that 'hello2.data2.subdata2' is included as a package by using
      the `packages` configuration field or the proper discovery methods
      (for example by using `find_namespace_packages(...)`/`find_namespace:`
      instead of `find_packages(...)`/`find:`).
  
      You can read more about "package discovery" and "data files" on setuptools
      documentation page.
  
  
  !!
  
    check.warn(importable)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================================================== short test summary info ========================================================================================
SKIPPED [1] tests/test_distribution.py:12: pytest_virtualenv not available. See #228
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! _pytest.outcomes.Exit: Setuptools_scm installed and nosetuptoolsscm tests not skipped. !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=============================================================== 1 failed, 57 passed, 1 skipped, 12 warnings, 4 errors in 84.73s (0:01:24) ===============================================================
```

**Comment by yurivict** [comment]
Tests for the version 0.16.7 now fail like this:
```
cd /usr/ports/devel/py-scikit-build/work-py39/scikit-build-0.16.7 && /usr/bin/env XDG_DATA_HOME=/usr/ports/devel/py-scikit-build/work-py39  XDG_CONFIG_HOME=/usr/ports/devel/py-scikit-build/work-py39  XDG_CACHE_HOME=/usr/ports/devel/py-scikit-build/work-py39/.cache  HOME=/usr/ports/devel/py-scikit-build/work-py39 PATH=/usr/local/libexec/ccache:/usr/ports/devel/py-scikit-build/work-py39/.bin:/home/yuri/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin PKG_CONFIG_LIBDIR=/usr/ports/devel/py-scikit-build/work-py39/.pkgconfig:/usr/local/libdata/pkgconfig:/usr/local/share/pkgconfig:/usr/libdata/pkgconfig MK_DEBUG_FILES=no MK_KERNEL_SYMBOLS=no SHELL=/bin/sh NO_LINT=YES PREFIX=/usr/local  LOCALBASE=/usr/local  CC="cc" CFLAGS="-O2 -pipe  -fstack-protector-strong -fno-strict-aliasing "  CPP="cpp" CPPFLAGS=""  LDFLAGS=" -fstack-protector-strong " LIBS=""  CXX="c++" CXXFLAGS="-O2 -pipe -fstack-protector-strong -fno-strict-aliasing  "  MANPREFIX="/usr/local" CCACHE_DIR="/tmp/.ccache" BSD_INSTALL_PROGRAM="install  -s -m 555"  BSD_INSTALL_LIB="install  -s -m 0644"  BSD_INSTALL_SCRIPT="install  -m 555"  BSD_INSTALL_DATA="install  -m 0644"  BSD_INSTALL_MAN="install  -m 444" /usr/local/bin/python3.9 -m pytest -k '' -rs -v -o addopts= 
========================================================================================== test session starts ==========================================================================================
platform freebsd13 -- Python 3.9.16, pytest-7.2.0, pluggy-1.0.0 -- /usr/local/bin/python3.9
cachedir: .pytest_cache
Using --randomly-seed=2205959646
installed packages of interest: build==0.10.0 distro==1.8.0 packaging==21.3 pip==22.3.1 setuptools==63.1.0 setuptools_scm==6.4.2 virtualenv==20.17.0 wheel==0.38.4
rootdir: /usr/ports/devel/py-scikit-build/work-py39/scikit-build-0.16.7, configfile: pyproject.toml, testpaths: tests
plugins: flake8-1.1.1, timeout-2.1.0, randomly-3.12.0, forked-1.4.0, rerunfailures-10.1, flaky-3.7.0, xdist-2.5.0, env-0.6.2, mock-3.10.0, freezegun-0.4.2, cov-4.0.0, nbval-0.10.0
collected 240 items / 1 skipped                                                                                                                                                                         
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/_pytest/main.py", line 270, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/_pytest/main.py", line 323, in _main
INTERNALERROR>     config.hook.pytest_collection(session=session)
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/pluggy/_hooks.py", line 265, in __call__
INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/pluggy/_manager.py", line 80, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/pluggy/_callers.py", line 60, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/pluggy/_result.py", line 60, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/pluggy/_callers.py", line 39, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/_pytest/main.py", line 334, in pytest_collection
INTERNALERROR>     session.perform_collect()
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/_pytest/main.py", line 667, in perform_collect
INTERNALERROR>     hook.pytest_collection_modifyitems(
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/pluggy/_hooks.py", line 265, in __call__
INTERNALERROR>     return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/pluggy/_manager.py", line 80, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/pluggy/_callers.py", line 60, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/pluggy/_result.py", line 60, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/pluggy/_callers.py", line 39, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/pytest_freezegun.py", line 55, in pytest_collection_modifyitems
INTERNALERROR>     if get_closest_marker(item, MARKER_NAME):
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/pytest_freezegun.py", line 17, in get_closest_marker
INTERNALERROR>     if LooseVersion(pytest.__version__) < LooseVersion('3.6.0'):
INTERNALERROR>   File "/usr/local/lib/python3.9/site-packages/setuptools/_distutils/version.py", line 55, in __init__
INTERNALERROR>     warnings.warn(
INTERNALERROR> DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.

========================================================================================== 1 skipped in 0.89s ===========================================================================================
*** Error code 3
```

Python-3.9
FreeBSD 13.1

**Comment by henryiii** [comment]
Looks like `pytest_freezegun` (used by pluggy) is abandoned. There's a maintained "pytest-freezer" replacement. I have no idea what's pulling in `pytest_freezegun`, though.

**Comment by henryiii** [comment]
I don't see anything in our dependencies (including `[test]`) that requires `pytest-freezegun`:

```
build==0.9.0
  - packaging [required: >=19.0, installed: 21.3]
    - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
  - pep517 [required: >=0.9.1, installed: 0.13.0]
codecov==2.1.12
  - coverage [required: Any, installed: 6.5.0]
  - requests [required: >=2.7.9, installed: 2.28.1]
    - certifi [required: >=2017.4.17, installed: 2022.9.24]
    - charset-normalizer [required: >=2,<3, installed: 2.1.1]
    - idna [required: >=2.5,<4, installed: 3.4]
    - urllib3 [required: >=1.21.1,<1.27, installed: 1.26.13]
Cython==0.29.32
flake8==6.0.0
  - mccabe [required: >=0.7.0,<0.8.0, installed: 0.7.0]
  - pycodestyle [required: >=2.10.0,<2.11.0, installed: 2.10.0]
  - pyflakes [required: >=3.0.0,<3.1.0, installed: 3.0.1]
pip==22.3
pipdeptree==2.5.2
pytest-cov==4.0.0
  - coverage [required: >=5.2.1, installed: 6.5.0]
  - pytest [required: >=4.6, installed: 7.2.0]
    - attrs [required: >=19.2.0, installed: 22.1.0]
    - iniconfig [required: Any, installed: 1.1.1]
    - packaging [required: Any, installed: 21.3]
      - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
    - pluggy [required: >=0.12,<2.0, installed: 1.0.0]
pytest-mock==3.10.0
  - pytest [required: >=5.0, installed: 7.2.0]
    - attrs [required: >=19.2.0, installed: 22.1.0]
    - iniconfig [required: Any, installed: 1.1.1]
    - packaging [required: Any, installed: 21.3]
      - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
    - pluggy [required: >=0.12,<2.0, installed: 1.0.0]
pytest-runner==6.0.0
pytest-virtualenv==1.7.0
  - pytest [required: Any, installed: 7.2.0]
    - attrs [required: >=19.2.0, installed: 22.1.0]
    - iniconfig [required: Any, installed: 1.1.1]
    - packaging [required: Any, installed: 21.3]
      - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
    - pluggy [required: >=0.12,<2.0, installed: 1.0.0]
  - pytest-fixture-config [required: Any, installed: 1.7.0]
    - pytest [required: Any, installed: 7.2.0]
      - attrs [required: >=19.2.0, installed: 22.1.0]
      - iniconfig [required: Any, installed: 1.1.1]
      - packaging [required: Any, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - pluggy [required: >=0.12,<2.0, installed: 1.0.0]
  - pytest-shutil [required: Any, installed: 1.7.0]
    - contextlib2 [required: Any, installed: 21.6.0]
    - execnet [required: Any, installed: 1.9.0]
    - mock [required: Any, installed: 4.0.3]
    - path.py [required: Any, installed: 12.5.0]
      - path [required: Any, installed: 16.5.0]
    - pytest [required: Any, installed: 7.2.0]
      - attrs [required: >=19.2.0, installed: 22.1.0]
      - iniconfig [required: Any, installed: 1.1.1]
      - packaging [required: Any, installed: 21.3]
        - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
      - pluggy [required: >=0.12,<2.0, installed: 1.0.0]
    - six [required: Any, installed: 1.16.0]
    - termcolor [required: Any, installed: 2.1.1]
  - virtualenv [required: Any, installed: 20.17.0]
    - distlib [required: >=0.3.6,<1, installed: 0.3.6]
    - filelock [required: >=3.4.1,<4, installed: 3.8.0]
    - platformdirs [required: >=2.4,<3, installed: 2.5.4]
scikit-build==0.16.3.dev10+g6364ea9.d20221130
  - distro [required: Any, installed: 1.8.0]
  - packaging [required: Any, installed: 21.3]
    - pyparsing [required: >=2.0.2,!=3.0.5, installed: 3.0.9]
  - setuptools [required: >=42.0.0, installed: 65.5.0]
  - wheel [required: >=0.32.0, installed: 0.37.1]
ubelt==1.2.2
xdoctest==1.1.0
  - six [required: Any, installed: 1.16.0]
```

**Comment by yurivict** [comment]
Closing as a stale issue.
